java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/obf/OsmGpxWriteContext.java

```diff
@@ -129,112 +138,173 @@ public void writeTrack(OsmGpxFile gpxInfo, Map<String, String> extraTrackTags, G
 		List<String> pgIcons = new ArrayList<>();
 		List<String> pgColors = new ArrayList<>();
 		List<String> pgBackgrounds = new ArrayList<>();
 		for (String name : pointsGroups.keySet()) {
-			pgNames.add(name);
-			pgIcons.add(pointsGroups.get(name).getIconName());
-			pgBackgrounds.add(pointsGroups.get(name).getBackgroundType());
-			pgColors.add(KAlgorithms.INSTANCE.colorToString(pointsGroups.get(name).getColor()));
-		}
-		final String delimiter = OBF_POINTS_GROUPS_DELIMITER;
-		gpxTrackTags.put(OBF_POINTS_GROUPS_NAMES, String.join(delimiter, pgNames));
-		gpxTrackTags.put(OBF_POINTS_GROUPS_ICONS, String.join(delimiter, pgIcons));
-		gpxTrackTags.put(OBF_POINTS_GROUPS_COLORS, String.join(delimiter, pgColors));
-		gpxTrackTags.put(OBF_POINTS_GROUPS_BACKGROUNDS, String.join(delimiter, pgBackgrounds));
+			PointsGroup pg = pointsGroups.get(name);
+			if (pg.getIconName() != null || pg.getBackgroundType() != null || pg.getColor() != 0) {
+				pgNames.add(name);
+				pgIcons.add(pg.getIconName());
+				pgBackgrounds.add(pg.getBackgroundType());
+				pgColors.add(KAlgorithms.INSTANCE.colorToString(pg.getColor()));
+			}
+		}
+		if (!pgNames.isEmpty()) {
+			final String delimiter = OBF_POINTS_GROUPS_DELIMITER;
+			gpxTrackTags.put(OBF_POINTS_GROUPS_NAMES, String.join(delimiter, pgNames));
+			gpxTrackTags.put(OBF_POINTS_GROUPS_ICONS, String.join(delimiter, pgIcons));
+			gpxTrackTags.put(OBF_POINTS_GROUPS_COLORS, String.join(delimiter, pgColors));
+			gpxTrackTags.put(OBF_POINTS_GROUPS_BACKGROUNDS, String.join(delimiter, pgBackgrounds));
+		}
 	}
 
-	private void addExtensionsTags(Map<String, String> gpxTrackTags, Map<String, String> extensions) {
-		if (extensions != null) {
+	private void addExtensionsTags(Map<String, String> gpxTrackTags, Map<String, String> extensions, OsmGpxFile gpxInfo) {
+		final String[] asIsTagsByPrefix = {
+				"gpx_",
+				"osm_",
+				"shield_",
+				"relation_gpx",
+				"color", // will be transformed to color_$color during the OBF-generation process
+				// "ref", "name", "description", and some other generic tags might be already added w/o prefix
+		};
+		if (extensions != null && !extensions.isEmpty()) {
 			if (extensions.containsKey("color")) {
-				gpxTrackTags.put("color", extensions.get("color")); // osmand:color
 				// prioritize osmand:color over GPX color
 				gpxTrackTags.remove("colour_int");
 				gpxTrackTags.remove("colour");
 			}
-			if (extensions.containsKey("width")) {
-				gpxTrackTags.put("width", extensions.get("width")); // osmand:width
+			if (extensions.containsKey("shield_text")) {
+				gpxInfo.updateRef(extensions.get("shield_text")); // update from osmc_text
+				gpxTrackTags.put("ref", gpxInfo.getPrettyRef()); // "osm_ref" will be kept as is
 			}
 			for (String key : extensions.keySet()) {
-				gpxTrackTags.put(OBF_GPX_EXTENSION_TAG_PREFIX + key, extensions.get(key));
+				String outputKey = OBF_GPX_EXTENSION_TAG_PREFIX + key; // gpx_ by default
```

Comment:
I think we shouldn't use extensions at all except some exceptions which could list that could be used for specific rendering purposes like "gpx_width", "gpx_color" in that case we won't need asIsTagsByPrefix

Reply:
`asIsTagsByPrefix` and tag prefixes refactored, Map-section is not filled with osm-tags now, POI-section is filled with original osm-tags without prefixes (only for Routes, not for Points)